Automattic/wp-calypso
# Add/show validation notice for unvalidated recovery emails

## Summary
This PR adds the unvalidated warning notice right below the recovery email if it is not unvalidated.
![image](https://cloud.githubusercontent.com/assets/1842898/21304670/d22c05ba-c601-11e6-8c25-3c275a70aac4.png), along with a "resend" button.

Before this PR, we don't have a way to validate the ownership / correctness of recovery emails entered by users, causing them to lose a reliable option for account recovery. 

## Test plan
Before testing, please make sure you have deployed D3185.

1. Log in as a user with a unvalidated recovery email set.
    1. Visit http://calypso.localhost:3000/me/security/checkup , and see the unvalidated warning notice.
    1. Click "resend", and check your mailbox to find the validation email
    1. Click the "Confirm" button, and your recovery email should be validated successfully.
    1. Refresh the security checkup page, and see the unvalidated warning gone.
1. Log in as a user without a recovery email set
    1. The unvalidated warning notice shouldn't be there.
    1. Add a recovery email.
    1. Check your mailbox, you should receive a validation email before hitting the resend button.
1. Log in as a user with a validated recovery email set.
    1. The unvalidated warning notice shouldn't be there.
    1. Update the recovery email.
    1. Check your mailbox, you should receive a validation email.
